gateway-ha/src/test/java/io/trino/gateway/ha/HaGatewayTestUtils.java

```diff
@@ -159,6 +142,29 @@ public static void setUpBackend(
                         .build();
         Response response = httpClient.newCall(request).execute();
         assertThat(response.isSuccessful()).isTrue();
+        TrinoHealthStateType newClusterHealthState = TrinoHealthStateType.PENDING;
+        long startTime = System.currentTimeMillis();
+        // pull cluster health states for 10 seconds
+        // It should be enough as the healthcheck is run every second
+        int timeout = 10 * 1000;
+        while (newClusterHealthState != TrinoHealthStateType.HEALTHY && (System.currentTimeMillis() - startTime) < timeout) {
+            // check the state of newly added cluster every second
+            request = new Request.Builder()
+                    .url(String.format("http://localhost:%s/api/public/backends/%s/state", routerPort, name))
+                    .get()
+                    .build();
+            response = httpClient.newCall(request).execute();
+            if (response.isSuccessful()) {
+                JsonCodec<ClusterStats> responseCodec = JsonCodec.jsonCodec(ClusterStats.class);
+                ClusterStats clusterStats = responseCodec.fromJson(response.body().string());
+                newClusterHealthState = clusterStats.healthState();
+                log.info("health state for trino cluster %s is %s", name, newClusterHealthState);
+            }
+            else {
+                Thread.sleep(1000);
+            }
+        }
+        assertThat(newClusterHealthState).isEqualTo(TrinoHealthStateType.HEALTHY);
```

Comment:
This is a utility class that manages helper methods. I don't understand why this check & assertion exist here. Could you extract into a dedicated test?

Reply:
Not sure if I follow. This check and assert is added to make sure when `setupBackend` is called, the backends state are healthy before returning to the caller.